File: leet/0058-length-of-last-word/length-of-last-word.go
lengthOfLastWord(s string) int {
    
    if len(s) == 0 {return 0}
    
    length := len(s)
    lastResult := 0
    result := 0
    
    for i := 0; i < length; i++ {
        if s[i] == ' ' {
            if result != 0 {
                lastResult = result
                result = 0
            }
        } else {
            result += 1
        }
    }
    
    if result == 0 {return lastResult}
    return result
}

File: leet/0058-length-of-last-word/length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    if (s.length == 0) return 0
    let i = s.length - 1
    while(i>= 0 && s[i] == " ") i--
    if (i < 0) return 0
    let ret = 0
    while(i >= 0 && s[i] != " ") { ret++; i-- }
    return ret
};
